### ISSUE TITLE
load() and multiple remote resources with dependencies

### ISSUE THREAD

**Issue by Zipelkopf** [jquery/jquery#4213]
### Description ###
Hi,
I'm using the JQuery load() and I have noticed an issue/behavior that perhaps could be improved.
```
var targetUrl = "xxx/ddd/page.html";
$("#container").load(targetUrl );
```
```
<div>This is the Target URL to be loaded</div>
<script src="js/library.js"></script>
<script src="js/script.js"></script>
```

If the **library.js** contains the definition of functions and the **script.js** contains the _$(document).ready_, then the _$(document).ready_ will not wait for **library.js** to be loaded.

```
// library.js
something = {};
something.myFunction = function(param) {
  alert(param);
};
```
```
// script.js
$(document).ready(function () {
  something.myFunction("Hello  World");
});
```
**Randomly**, I get the error message saying that _myFunction_ is not defined.
Would it be possible to extend the **load()** to allow the  _$(document).ready_ in the **script.js** to take into considerations the other resources? Or alternatively to tell the load() to fetch and process the resources in the target URL in the same order they are found in the page, perhaps in a deferred way?

Otherwise, what are the alternatives in this scenario?
Thanks a lot & Regards

**Comment by timmywil** [comment]
Thanks for opening an issue. Could you make a small example? Scripts should be executed synchronously.

**Comment by Zipelkopf** [comment]
Hi, 
please find attached the test case.

[jquery.load.zip](https://github.com/jquery/jquery/files/2542749/jquery.load.zip)

I have put together a simple test case and I noticed that the resources where correctly loaded in a sync way... and I noticed as well the following console message:

> [Deprecation] Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. 

This console message does not appear on my application, then I looked at it and I realized the following lines of code where introduced:

```
$.ajaxPrefilter(function (options, originalOptions, jqXHR) {
    options.async = true;
}); 
```

As soon as I set back the **options.async = false;** the load() started working normally.
Please note the issue happens only at the first load, with no cached scripts.

![capture](https://user-images.githubusercontent.com/21308710/47920077-0cc6e200-dea9-11e8-9de6-e253942036a9.JPG)

Considering that the majority of the Ajax calls will be Async, is it possible to pass a parameter to the load() to instruct loading the remote resources in a sync way? I tried replacing the load() with the get() or with $.ajax by passing explicitly async = false, but it does not work when the above options.async is set to true.

Long story short, I have to remove the options.async = true, but I wonder if I can still run the majority of my Ajax calls in a Async way and process only the remote resources fetched with load() in a synch one.

Thanks a mill for the support.

**Comment by timmywil** [comment]
Ok, in that case, if you need to keep the prefilter that sets async to true, you can add another prefilter that sets async back to false–as long as it executes after the existing prefilter. For example (untested),

```js
$.ajaxPrefilter(function (options, originalOptions, jqXHR) {
    if (/library/.test(options.url)) {
        options.async = false
    }
});
```

I think `.load` is working as expected so I'll close.

**Comment by Zipelkopf** [comment]
Thanks a lot! I'm going to play with your suggestion.
Regards
